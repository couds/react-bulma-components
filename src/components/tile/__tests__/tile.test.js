import React from 'react';
import { render } from 'react-testing-library';
import { Tile } from '..';
import CONSTANTS from '../../../constants';

describe('Tile component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <Tile>
        <img alt="placeholder" src="http://bulma.io/images/placeholders/128x128.png" />
      </Tile>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should concat classname in props with classname', () => {
    const { asFragment } = render(<Tile className="other-class this-is-a-test" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should use inline styles', () => {
    const { asFragment } = render(
      <Tile style={{ height: 250 }}>
        <p>Default</p>
      </Tile>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render as Section', () => {
    const { asFragment } = render(
      <Tile renderAs="section">
        <p>Default</p>
      </Tile>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  Object.values(CONSTANTS.COLORS)
    .filter(c => c)
    .map(color =>
      it(`Should use use color ${color}`, () => {
        const { container } = render(
          <Tile notification color={color}>
            <p>Default</p>
          </Tile>
        );
        expect(container.firstChild).toHaveClass(`is-${color}`);
      })
    );
});
