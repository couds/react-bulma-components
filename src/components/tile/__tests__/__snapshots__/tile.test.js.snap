// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <div
    class="tile other-class this-is-a-test"
  />
</DocumentFragment>
`;

exports[`Tile component should render 1`] = `
<DocumentFragment>
  <div
    class="tile"
  >
    <img
      alt="placeholder"
      src="http://bulma.io/images/placeholders/128x128.png"
    />
  </div>
</DocumentFragment>
`;

exports[`Tile component should render as Section 1`] = `
<DocumentFragment>
  <section
    class="tile"
  >
    <p>
      Default
    </p>
  </section>
</DocumentFragment>
`;

exports[`Tile component should use inline styles 1`] = `
<DocumentFragment>
  <div
    class="tile"
    style="height: 250px;"
  >
    <p>
      Default
    </p>
  </div>
</DocumentFragment>
`;
